### PR TITLE
Account creation

### PR DESCRIPTION
**Pull request by michding**
Model form can be implemented to ensure our model:
Model form 
https://docs.djangoproject.com/en/dev/topics/forms/modelforms/#modelform

Also implemented a password filter. 
Idea: Level the password such as Strong, Medium, Weak(Unacceptable)

### CODE REVIEW

File: code/books4u/api.py [hunk 1]
```diff
@@ -16,3 +16,24 @@ def get_all_books(request):
         })
 
     return HttpResponse(json.dumps(response_data), content_type="application/json")
+
+
+def create_account(request):
```
Comment (wzs1234566) [inline code/books4u/api.py]: need unit test on this

File: code/books4u/utils.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from .models import *
+import re
+
+
+def password_filter(pwd):
```
Comment (wzs1234566) [inline code/books4u/utils.py]: unit test this too

File: code/books4u/api.py [hunk 3]
```diff
@@ -61,3 +61,22 @@ def login(request):
...
+    if request.method == "POST":
+        account_name = request.POST.get('name', '')
+        account_password = request.POST.get('password')
+        # Password filter can be implemented here
+        account_e_mail = request.POST.get('e_mail')
+        if User.objects.filter(e_mail=account_e_mail).exists():
+            return HttpResponse(json.dumps({'status': 'failure', 'detail': 'existing_email'}), content_type="application/json")
```
Comment (wzs1234566) [inline code/books4u/api.py]: 这里统一用reason不用detail吧。

File: code/books4u/utils.py [hunk 4]
```diff
@@ -1,9 +1,19 @@
 from .models import *
+import re
 from django.contrib.sessions.backends.db import SessionStore
 from django.contrib.sessions.models import Session
 from django.utils import timezone
 
 
+def password_filter(pwd):
```
Comment (wzs1234566) [inline code/books4u/utils.py:8]: unit test this.
It seems like it was not used in api.py?

Comment (michding) [reply code/books4u/utils.py:8]: @wzs1234566  not implemented yet. I will talk to front end team about return info

File: code/books4u/api.py (hunk 3 above)
Comment (michding) [reply code/books4u/api.py]: Sure. Will change it during next commit

**Comment by wzs1234566** [comment]
Please add personal_intro field and gender field.
Also, there is this bug where you send a post request containing nothing, the registration will still succeed.

**Comment by michding** [comment]
@wzs1234566 
From what I learned from @Elsie4ever , our front end will never post if a constrain field is empty. So the main reason my test failed was because in my Testcase setup, I only created with pwd and email...

**Comment by michding** [comment]
@wzs1234566 Issues have been fixed.
Also added test cases for password filter

**Comment by michding** [comment]
"Also, there is this bug where you send a post request containing nothing, the registration will still succeed."
Since the front end will never send empty request, should I still set up anyways a condition to check if there is any empty field?
@wzs1234566

**Comment by michding** [comment]
capture
Added gender and it fails. Image above is the log captured during the test. Not sure if that not-null constraint is from our model, probably from django. 
It can be fixed by adding another parameter to request.POST.get, like what I did for gender earlier, which sets the detail to 'Unknown' instead of None.

**Comment by Elsie4ever** [comment]
Not exactly front end will never send empty request. In registration page, I set all attribute as required except "personal intro", because I think that it makes sense to add intro later on.

**Comment by michding** [comment]
@Elsie4ever I see. Will set personal intro default to empty then.

**Comment by wzs1234566** [comment]
@Elsie4ever @michding From my testing, The front end is able to send a all-empty request

**Comment by wzs1234566** [comment]
@michding Yes. Even front end is checking for empty fields, backend should still double check.

**Review by wzs1234566 — APPROVED** [review]

**Pull request merged**